README.md

```diff
@@ -11,15 +10,20 @@ Install cookiecutter.
 pip install cookiecutter
 ```
 
-Use it to generate a package, following prompts fill in the name and authorship
-of your new JupyterLab extension.
+Use cookiecutter to generate a package, following the prompts to fill in the name and authorship of your new JupyterLab extension.
 
 ```
 cookiecutter https://github.com/jupyter/jupyterlab-extension-cookiecutter
 ```
 
-A very simple working extension, written in common JavaScript, is included in
-the ``lib/`` directory. Use this example to build your own extension.
+## A simple example
 
-User and developer installation instructions are included in the README of the
-package generated by cookiecutter.
+The ``lib/`` directory of your new extension includes a very simple example of a working extension, written in JavaScript using CommonJS modules. Use this example as a guide to build your own extension.
+
+## Package names
+
+JupyterLab extensions generated by this cookie cutter recipe are composed of both a python package and an npm package. The npm package is declared as a private package, so it cannot be published to the public npm package repository. The npm package is used to retrieve npm dependencies and provide a framework for the bundling of the javascript. The python package is published to the PyPI python package repository, and contains the generated JupyterLab extension Javascript bundle.
+
+To lessen confusion between the different packages, this recipe sets the extension name, the python package name, and the npm package name all to the same thing. We also use the python package name as the python module name. A consequence of this is that the extension name should be a valid python import name (e.g., cannot contain dashes).
```

Comment:
To simplify the developer experience and lessen confusion between the different packages, the cookiecutter template unifies the extension name, the Python package name, and the npm package name to the same name. Make sure to use a valid Python name (i.e. cannot contain dashes) for the extension name since the extension is used as a Python module as well as a Python package.